public/services/ipv6-services.js

```diff
@@ -0,0 +1,21 @@
+export const compressIPv6 = (ip) => {
+  if (typeof (ip) !== 'string') {
+    return ip;
+  }
+  if (ip?.split(':').length !== 8) {
+    return ip;
+  }
+
+  let output = ip.split(':').map(terms => terms.replace(/\b0+/g, '') || '0').join(":");
+  let zeros = [...output.matchAll(/\b:?(?:0+:?){2,}/g)];
```

Comment:
Probably can be switched to `const`.

Also, would be great to migrate this service to TypeScript

Reply:
Changes implemented on c0f53f5b2a93db93599009d321f53c5d1b244699